[keymgr, doc] Update diversification inputs (#24396)
lowRISC/opentitan: doc/security/specs/identities_and_root_keys/README.md

    </td>
  </tr>
  <tr>
    <td>HardwareRevisionSecret</td>
    <td>Gates</td>
    <td>
Encoded in gates. Provisioned by Silicon Creator before tapeout. Hidden from
software.
    </td>
  </tr>
  <tr>
    <td id="rom-hash">RomHash</td>
    <td>Computed by ROM controller</td>
    <td>
SHA-3-256 hash of the ROM image.
    </td>
  </tr>
  <tr>
    <td>Health State Measurement</td>
    <td>Computed by Lifecycle controller</td>
    <td>
Comprises the following measurements:

* Device life cycle state.
* Debug mode state.

The debug mode shall be used as well if there are multiple debug configurations
supported by a single life cycle state.
    </td>
  </tr>
  <tr>
    <td>DeviceIdentifier</td>
    <td>OTP</td>
    <td>
Provisioned at manufacturing time. Readable from software and JTAG interface.
    </td>
  </tr>
  <tr>

>     <td>ROMExtSecurityDescriptor</td>

Reviewer: I think a different name should be used here. It is the SW binding of this stage correct?

Author: Renamed this to `SoftwareBindingValue` and update the description to capture it is either OTP measurement (for attestation) or ROM_EXT manifest value (for sealing).

    <td>SW register</td>
    <td>

> The implementation may choose one of the following options:

Reviewer: Our ROM implementation uses measurements of the OTP SW configuration partitions and the secure boot key partitions. See https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/rom.c;l=562?q=rom.c&ss=opentitan%2Fopentitan

Author: Updated the description to reflect that we are not using ROM_Ext measurement here, but OTP measurement instead.

> 1. Hash of the ROM extension. Changes in the ROM extension code will trigger an

Reviewer: Hash of ROM_EXT should be used for the OwnerIntermediateKey; we need CreatorRootKey to remain stable for lifetime of chip.

Author: Updated as part of the previous comment.

    <td>OTP</td>
    <td>
Provisioned at manufacturing time. Readable from software and JTAG interface.
    </td>
  </tr>
  <tr>
    <td>SoftwareBindingValue</td>
    <td>SW register</td>
    <td>
For the attestation chain, this register is populated with the hash of the OTP
image. Therefore, changes in the OTP configuration will trigger an update of
the CreatorIdentity for attestation.

For the sealing chain, this register uses a binding tag stored in the ROM_EXT
manifest. This is to retain the same CreatorIdentity across validated updates of
the ROM_EXT. This allows sealing keys to be persistent across ROM_EXT updates.
    </td>
  </tr>
</table>

The CreatorRootKey can be used to generate the CreatorIdentity and the
OwnerIntermediateKey described in the following sections.

## CreatorIdentity

The CreatorIdentity is an asymmetric key derived from the `CreatorRootKey`. It
is used as a cryptographic identifier bound to the device and the Silicon
Creator. It is used to attest to the authenticity of the physical device and the
ROM and OTP configuration.

CreatorIdentity becomes part of the UDS certificate and it is
generated as follows:

```
CreatorIdentitySeed =

>   KM_DERIVE(CreatorRootKey, OTBNDiversificationConstant)

Reviewer: Not sure what you are referring to by the `OTBNDiversificationConstant`, but if it is this the [diversification constant used by the output-generate function for CreatorIdentitySeed](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/lib/cert/dice.c;l=78?q=sc_keymgr_diversification_t&ss=opentitan%2Fopentitan), then I would not say it is specific to OTBN, it is specific to the output-generate invocation we are using for this specific UDS (CreatorIdentity) keygen, i.e., it is the `CreatorIdentityDiversifier` or something like that.

Reviewer: If you are [referring to this](https://cs.opensource.google/opentitan/opentitan/+/master:hw/ip/keymgr/rtl/keymgr.sv;l=30?q=otbn&ss=opentitan%2Fopentitan:hw%2Fip%2Fkeymgr%2F), then I would say that the KM_DERIVE needs an additional input then as well (the `CreatorIdentityDiversifier` I mentioned as well).

Author: Added `CreatorIdentityDiversifier` as an extra input.

## OwnerIntermediateKey

The `OwnerIntermediateKey` is used as an intermediate virtual entity between the
SiliconCreator and SiliconOwner. It is used to establish a cryptographic link

> to CreatorSeed as well as the integrity of ROM_Ext image.

Reviewer: ```suggestion
to CreatorSeed as well as the integrity of ROM_EXT image.
```

chain.

```
OwnerIntermediateIdentitySeed =
  KM_DERIVE(OwnerIntermediateKey, OTBNDiversificationConstant)

// ASYM_KDF is a KDF function compliant to the Asymmetric Key
// requirements defined in the Attestation specification document.
OwnerIntermediateIdentity_Private = ASYM_KDF(OwnerIntermediateIdentitySeed)
CLEAR_BEFORE_NEXT_BOOT_STAGE(OwnerIntermediateIdentitySeed, OwnerIntermediateIdentity_Private)
```

<table>
  <tr>
    <td><strong>Name</strong></td>
    <td><strong>Encoding</strong></td>
    <td><strong>Description</strong></td>
  </tr>
  <tr>
    <td>OwnerIntermediateIdentitySeed</td>
    <td>SW Register Output</td>
    <td>Seed used to generate the OwnerIntermediateIdentity asymmetric key.</td>
  </tr>
  <tr>
    <td>OTBNDiversificationConstant</td>
    <td>Gates</td>
    <td>
A constant defined in gates. Used to diversify sideloaded values to OTBN.

Hidden from software.
    </td>
  </tr>
</table>

The `OwnerIntermediateIdentitySeed` and the private portion of the
OwnerIntermediateIdentity shall be cleared before the bootloader (BL0) hands
over execution to the kernel.

## OwnerKey

The `OwnerKey` is the key diversified with `OwnerSeed` and it is associated

> with the CDI_1 certificate.

Reviewer: maybe you should say it also provides a cryptographic link to the Owner FW image and Owner Configuration Block?

> The `OwnerIntermediateKey` is generated as follows:

Reviewer: ```suggestion
The `OwnerKey` is generated as follows:
```

```
OwnerKey =
   KM_DERIVE(OwnerIntermediateKey, OwnerSeed | SoftwareBindingValue)
```

<table>
  <tr>
    <td><strong>Name</strong></td>
    <td><strong>Encoding</strong></td>
    <td><strong>Description</strong></td>
  </tr>
  <tr>
    <td> id="creator-secret"OwnerSeed</td>
    <td>Flash</td>
    <td>
Additional diversification key stored in flash. Provisioned during ownership
transfer.

Hidden from software once provisioned.
    </td>
  </tr>
  <tr>
    <td>SoftwareBindingValue</td>
    <td>SW Register<br>Lockable Input</td>
    <td>
For the attestation chain, this register is populated with owner-specific

> measurements.

Reviewer: it is just a measurement of the combination of the owner config block and owner fw.

For the sealing chain, this register uses a diversification value determined by

> owner.

Reviewer: The diversifier will be a combination of the application key domain (i.e., which key, from {prod,dev,test}, is used to verify the ROM_EXT to Owner firmware jump), and a diversifier specified by the owner configuration.

    </td>
  </tr>
</table>

## OwnerIdentity

The OwnerIdentity is used as a cryptographic identifier bound to the device and
the SiliconOwner. It is used in Attestation flows. OwnerIdentity is not

> expected to change during the lifetime of the device ownership assignment.

Reviewer: that is not true. It will change if owner firmware is updated, even if the owner stays the same

Author: Removed the last incorrect sentence.

    <td><strong>Description</strong></td>
  </tr>
  <tr>
    <td>OTBNDiversificationConstant</td>
    <td>Gates</td>
    <td>
A constant defined in gates. Used to diversify sideloaded values to OTBN.

Hidden from software.
    </td>
  </tr>
  <tr>
    <td>OwnerIdentitySeed</td>
    <td>SW Register Output</td>
    <td>Seed used to generate the OwnerIdentity asymmetric key.</td>
  </tr>
</table>

> The `OwnerIdentitySeed` and the private portion of the OwnerIdentity shall be
> cleared before the bootloader (BL0) hands over execution to the kernel.

Reviewer: I would just say:
```suggestion
The `OwnerIdentitySeed` and the private portion of the OwnerIdentity shall be
cleared after use by the Owner firmware.
```
